### PR TITLE
Future/peggy2 blocklist

### PR DESCRIPTION
**Pull request by daniportugalgit**
- [x] Add the blocklist to the contract suite
- [x] Add/update blocklist isolated unit tests
- [x] Update the blocklist as per the last changes on Peggy 1.0
- [x] Add Blocklist + BridgeBank + CosmosBridge unit tests
- [x] Review unused variables: are there any vars that are already deprecated and didn't make it to production?
- [x] Test upgrades with a mainnet fork and make sure they are safe and working.
- [x] Test Coverage: reach 100% coverage for all non-mock contracts (reached 100% in CosmosBrdige and 99.36% in BridgeBank, which is enough)
- [x] Re-enable the contract-sizer
- [x] Remove everything about the Ethereum Whitelist (as it's deprecated in Peggy2)
- [x] Add tests with Rowan (they were bundled together with EthWhitelist, that has been removed)
- [x] Bring Hardhat initialization goodies to this branch (logs, forking, colors)
- [x] Refactor BridgeBank (storage overlap)
- [x] Refactor CosmosBridge (storage overlap)
- [x] Script that performs the update from Peggy 1.0 to Peggy 2.0

TEST COVERAGE:
- [x] Re-enable test coverage with new tool that works with Peggy 2.0 codebase
- [x] Add a command to run the test coverage
- [x] Adjust the command to run tests
- [x] CosmosBridge: `onlyValidator()`
- [x] CosmosBridge: `changeOperator()`
- [x] BridgeBank: `getDenom()` when we already know the token's denom
- [x] BridgeBank: `getDenom()` when the contract does not have a `cosmosDenom` property
- [x] BridgeBank: `_burnTokens()` when `tokenAddress` is the old eRowan token (can't do this in a normal test, as we would need to deploy the erowan contract to the exact same address it has in production).
- [x] EthereumWhitelist: `onlyEthTokenWhiteList()` (remove it as it's deprecated)

BUGS:
- [x] Fix bug in which the BridgeBank deploys new tokens but doesn't mint
- [x] Add unit tests to guarantee that whenever BridgeBank deploys a new token, it also gets minted correctly

### CODE REVIEW

File: smart-contracts/contracts/BridgeBank/BridgeBank.sol [hunk 1]
```diff
@@ -10,794 +10,787 @@ import "./BankStorage.sol";
 import "./Pausable.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
+import "hardhat/console.sol";
```
Comment (daniportugalgit) [inline smart-contracts/contracts/BridgeBank/BridgeBank.sol:13]: Remove this

File: smart-contracts/contracts/BridgeBank/CosmosBankStorage.sol [hunk 2]
```diff
@@ -6,45 +6,39 @@ pragma solidity 0.8.0;
...
+
+	/**
+	 * @dev {DEPRECATED}
+	 */
+	uint256 private cosmosDepositNonce;
+
+	/**
+	 * @dev {DEPRECATED}
+	 */
+	mapping(string => address) internal controlledBridgeTokens;
```
Comment (daniportugalgit) [inline smart-contracts/contracts/BridgeBank/CosmosBankStorage.sol:33]: Shouldn't this be private?

File: smart-contracts/.prettierrc [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{
+  "overrides": [
+    {
+      "files": "*.sol",
+      "options": {
+        "printWidth": 100,
+        "tabWidth": 2,
+        "useTabs": true,
```
Comment (daniportugalgit) [inline smart-contracts/.prettierrc:8]: Change to spaces

File: smart-contracts/.prettierrc [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        "bracketSpacing": true,
+        "explicitTypes": "always"
+      }
+    },
+    {
+      "files": "*.js",
+      "options": {
+        "printWidth": 100,
+        "tabWidth": 2,
+        "useTabs": true,
```
Comment (daniportugalgit) [inline smart-contracts/.prettierrc:19]: change to spaces

File: smart-contracts/contracts/BridgeBank/BridgeBank.sol [hunk 5]
```diff
@@ -16,788 +16,779 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
...
+		uint256 amount
+	) internal {
+		// Transfer funds to intended recipient
+		if (token == address(0)) {
+			(bool success, ) = recipient.call{ value: amount }("");
+			require(success, "error sending ether");
+		} else {
+			IERC20 tokenToTransfer = IERC20(token);
+
+			// TODO: try/catch
```
Comment (daniportugalgit) [inline smart-contracts/contracts/BridgeBank/BridgeBank.sol:689]: remove this!

File: smart-contracts/scripts/upgrades/peggy2-2022-02/run.js [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+
+	// Get signers to send transactions
+	await setupAccounts();
+
+	// Fetch current values from the deployed contract
+	await setBridgeBankStorageSlots();
+	await setCosmosBridgeStorageSlots();
+
+	// Pause the system
+	await pauseBridgeBank();
```
Comment (daniportugalgit) [inline smart-contracts/scripts/upgrades/peggy2-2022-02/run.js:100]: Make it re-runable (check if it's paused, etc)

Comment (daniportugalgit) [reply smart-contracts/scripts/upgrades/peggy2-2022-02/run.js:100]: Pause before getting storage values

File: smart-contracts/test/testBridgeBank.js [hunk 7]
```diff
@@ -169,6 +171,55 @@ describe("Test Bridge Bank", function () {
...
+        state.sender,
+        bridgeToken.address,
+        state.amount
+      );
+
+      afterUserBalance = Number(await bridgeToken.balanceOf(userOne.address));
+      afterUserBalance.should.be.bignumber.equal(0);
+    });
+
+    it("should allow a user to burn a token that doesn't have a denom", async function () {
```
Comment (daniportugalgit) [inline smart-contracts/test/testBridgeBank.js:206]: No. Should not be able to burn a token without a denom string.
*make sure erowan still works!

**Review by banshee — APPROVED** [review]

**Review by Brando753 — APPROVED** [review]

**Pull request merged**